### ISSUE TITLE
Duotone primary/secondary doesn't feel consistent

### ISSUE THREAD

**Issue by martijnhartlief** [FortAwesome/Font-Awesome#15322]
**Describe the bug**
The primary/secondary part from duo-tone doesn't feel consistent, which makes some icons less visibile than others.
When an icon has something inside, the inside is now the primary color and the outside is the secondary color. "badge" visualizes the problem the most obviously. Creating a visual disparity between the original solid and the duo-tone variants.

Don't get me wrong, it looks cool on the font awesome overview, but it makes duo-tone impossible as a drop-in replacement for existing icons.

Ofcourse you can manually fix every icon by checking them one by one and change the css to match it, but it makes more sense for me if it was by default.

**To Reproduce**
badge (Background primary, inside secondary)
badge-check, badge-dollar, badge-percent (symbol primary, background secondary)

clone, folder, cloud (foreground primary)
paste, folders, cloud-download (background primary)

**Expected behavior**
I would expect the same type of icon have the same element being primary, instead of swapping it when a symbol is added.

**Version and implementation**
5.10
Chrome latest

- [x] SVG with JS
- [ ] Web Fonts with CSS
- [ ] SVG Sprites
- [ ] On the Desktop

**Bug report checklist**
- [x] I have filled out as much of the above information as I can
- [x] I have included a test case because my odds go _way_ up that the team can fix this when I do
- [x] I have [searched for existing issues](https://github.com/FortAwesome/Font-Awesome/issues) and to the best of my knowledge this is not a duplicate

**Comment by tagliala** [comment]
Hi!

Thanks for being part of the Font Awesome Community and thanks for reporting this.

@sensibleworld could you please take a look?

**Comment by sensibleworld** [comment]
Thanks for the feedback, @martijnhartlief! I'm actually working through some of these issue for our next release. For the badge icon, we'll most likely drop the interior stroke and revert the whole shape to `.secondary`. Would that solve it for you?

**Comment by InsanityMeetsHH** [comment]
If you compare [badge](https://fontawesome.com/icons/badge?style=duotone) with [certificate](https://fontawesome.com/icons/certificate?style=duotone), certificate has no secondary layer.

**Comment by martijnhartlief** [comment]
> Thanks for the feedback, @martijnhartlief! I'm actually working through some of these issue for our next release. For the badge icon, we'll most likely drop the interior stroke and revert the whole shape to `.secondary`. Would that solve it for you?

To be honest that would it make it worse... because that indeed solves the inconsistency between badge element, but it only worsens the effect of duotone, because enabling the duotone style makes the element 50% transparent by default. Imho it would make way more sense that the duotone behaves exactly like the solid version with the new duotone part being secondary. 

Some icons have the main part as primary which keeps the contrast in tact, that's what I would expect, like folder.

But making duotone icons fall back to secondary by default, and having the highlights as primary changes the whole contrast of the icon, making icons like for example receipt and badge almost not legible by default when placing it on a dark background.

**Comment by davegandy** [comment]
So the fun thing here is that we're using CSS variables for both opacity and color. Basically, anything you want to do, you can do in some interesting ways with CSS variables. We're polishing up consistency between a lot of these icons, but you're right. We're going to largely want the main portion of the icon to be transparent. That's not going to change. But this thread got us looking more into the `fa-swap-opacity` class, and we're going to try to get this inheriting as well, so you can set it on a div and get it working for a whole block of icons. Here's a prototype of what we're looking to add:
```
.fa-swap-opacity .fad:before {
  opacity: .4;
  opacity: var(--fa-secondary-opacity,.4);
}

.fa-swap-opacity .fad:after {
  opacity: 1;
  opacity: var(--fa-primary-opacity,1);
}
```

As for color, you can do the same thing with our variables. And you could put it all together and get your `fa-swap-duotone` class that would swap both opacity and any colors you've set. 

We're still definitely open to thinking about this. Do you have this in action anywhere? If we could see it, maybe we'd think about making the change you're suggesting.

**Comment by davegandy** [comment]
Heads up. Just released 5.10.1 last week. Based on this issue, we made it where `fa-swap-opacity` will work on individual icons as well as a block of them. (Basically, we added the code mentioned above.)

Leaving open for a bit more in case others have thoughts.
